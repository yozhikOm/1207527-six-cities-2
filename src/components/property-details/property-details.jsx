import React from 'react';
import PropTypes from 'prop-types';
import {Header} from '../header/header.jsx';
import {ReviewsList} from '../reviews-list/reviews-list.jsx';
import {Map} from '../map/map.jsx';
import {PropertyCard} from '../property-card/property-card.jsx';

const PropertyDetails = (props) => {
  const {currentCityCoords, offer, neighboringOffers, activeItemID, setActiveItem} = props;

  const offersArrayForMap = neighboringOffers.map((neibOffer) => (
    {
      id: neibOffer.id,
      coordinates: [neibOffer.location.latitude, neibOffer.location.longitude],
    })).concat([{
    id: offer.id,
    coordinates: [offer.location.latitude, offer.location.longitude],
  }]);

  return (
    <React.Fragment>
      <Header/>
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {offer.images.map((it, i) => {
                return (
                  <div key={`photo-${i}`} className="property__image-wrapper">
                    <img className="property__image" src={it} alt="Photo studio" />
                  </div>
                );
              })}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {offer.isPremium ?
                <div className="place-card__mark">
                  <span>Premium</span>
                </div> :
                <React.Fragment/>
              }
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {offer.title}
                </h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `96%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{offer.rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  Entire place
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {offer.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {offer.maxAdults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {offer.goods.map((it, i) => {
                    return (
                      <li key={`goods-${i}`} className="property__inside-item">
                        {it}
                      </li>
                    );
                  })}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img className="property__avatar user__avatar" src={offer.host.avatarUrl} width="74" height="74" alt="Host avatar" />
                  </div>
                  <span className="property__user-name">
                    {offer.host.name}
                  </span>
                  {offer.host.isPro ?
                    <span className="property__user-status">
                      Pro
                    </span> :
                    <React.Fragment/>
                  }

                </div>
                <div className="property__description">
                  <p className="property__text">
                    {offer.description}
                  </p>
                </div>
              </div>
              <ReviewsList offerId={offer.id} />
            </div>
          </div>
          <section className="property__map map">
            <Map
              currentCityCoords={currentCityCoords}
              offersArray={offersArrayForMap}
              activeItemID={activeItemID === -1 ? offer.id : activeItemID}
            />
          </section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              {neighboringOffers.map((item) => (
                <React.Fragment key={item.id}>
                  <PropertyCard offer={item} cardMouseEnterHandler={setActiveItem} />
                </React.Fragment>
              ))}
            </div>
          </section>
        </div>
      </main>
    </React.Fragment>
  );
};

PropertyDetails.propTypes = {
  currentCityCoords: PropTypes.arrayOf(PropTypes.number.isRequired, PropTypes.number.isRequired).isRequired,
  offer: PropTypes.shape({
    id: PropTypes.number.isRequired,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number,
      }),
    }).isRequired,
    previewImage: PropTypes.string,
    images: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool,
    isPremium: PropTypes.bool,
    rating: PropTypes.number,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number,
    maxAdults: PropTypes.number,
    price: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      isPro: PropTypes.bool,
      avatarUrl: PropTypes.string,
    }),
    description: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
  }),
  neighboringOffers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number,
      }),
    }).isRequired,
    previewImage: PropTypes.string,
    images: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool,
    isPremium: PropTypes.bool,
    rating: PropTypes.number,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number,
    maxAdults: PropTypes.number,
    price: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      isPro: PropTypes.bool,
      avatarUrl: PropTypes.string,
    }),
    description: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
  })),
  activeItemID: PropTypes.number,
  setActiveItem: PropTypes.func,
};

export {PropertyDetails};
