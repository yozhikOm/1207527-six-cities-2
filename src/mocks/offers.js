export const offers = [
  {
    "id": 1,
    "title": `Cozy house amsterdam`,
    "type": `House`,
    "price": 650,
    "photos": [
      {
        "src": `img/apartment-01.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.85309666406198],
  },
  {
    "id": 2,
    "title": `Cosy room in spacious apartment`,
    "type": `Room`,
    "price": 45,
    "photos": [
      {
        "src": `img/apartment-02.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.369553943508, 4.85309666406198],
  },
  {
    "id": 3,
    "title": `Elegant private room in stylish house`,
    "type": `Room`,
    "price": 128,
    "photos": [
      {
        "src": `img/room.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.929309666406198],
  },
  {
    "id": 4,
    "title": `Huge Tinyhouse in the green, between sea and city`,
    "type": `House`,
    "price": 1520,
    "photos": [
      {
        "src": `img/apartment-03.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3809553943508, 4.939309666406198],
  }
];
