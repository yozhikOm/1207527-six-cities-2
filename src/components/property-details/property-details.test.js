import React from 'react';
import {PropertyDetails} from './property-details.jsx';
import renderer from 'react-test-renderer';
import {offers} from '../../mocks/offers.js';

it(`PropertyDetails компонент рендерится корректно`, () => {
  const propertyDetailsComponent = renderer.create(<PropertyDetails {...offers[0]}/>).toJSON();
  expect(propertyDetailsComponent).toMatchSnapshot();
});
