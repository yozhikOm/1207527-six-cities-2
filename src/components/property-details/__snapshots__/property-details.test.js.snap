// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetails компонент рендерится корректно 1`] = `
<React.Fragment>
  <Header />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/4.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <React.Fragment />
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Penthouse, 4-5 rooms + 5 balconies
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "96%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3.6
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Entire place
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              5
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              459
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.
              </p>
            </div>
          </div>
          <ReviewsList
            offerId={1}
          />
        </div>
      </div>
      <section
        className="property__map map"
      >
        <Map
          activeItemID={-1}
          currentCityCoords={
            Array [
              50.938361,
              6.959974,
            ]
          }
          offersArray={
            Array [
              Object {
                "coordinates": Array [
                  50.869557,
                  4.332697,
                ],
                "id": 2,
              },
              Object {
                "coordinates": Array [
                  51.216402,
                  6.758314,
                ],
                "id": 3,
              },
              Object {
                "coordinates": Array [
                  50.957361,
                  6.9509739999999995,
                ],
                "id": 1,
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <React.Fragment>
            <PropertyCard
              offer={
                Object {
                  "bedrooms": 5,
                  "city": Object {
                    "location": Object {
                      "latitude": 50.846557,
                      "longitude": 4.351697,
                      "zoom": 13,
                    },
                    "name": "Brussels",
                  },
                  "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                  "goods": Array [
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 2,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/16.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/10.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/13.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/3.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 50.869557,
                    "longitude": 4.332697,
                    "zoom": 16,
                  },
                  "maxAdults": 10,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/13.jpg",
                  "price": 474,
                  "rating": 2.5,
                  "title": "Waterfront with extraordinary view",
                  "type": "house",
                }
              }
            />
          </React.Fragment>
          <React.Fragment>
            <PropertyCard
              offer={
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Dusseldorf",
                  },
                  "description": "Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.",
                  "goods": Array [
                    "Towels",
                    "Laptop friendly workspace",
                    "Baby seat",
                    "Washer",
                    "Air conditioning",
                    "Breakfast",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 3,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/10.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/13.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/9.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/11.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 51.216402,
                    "longitude": 6.758314,
                    "zoom": 16,
                  },
                  "maxAdults": 5,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/16.jpg",
                  "price": 474,
                  "rating": 2.4,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                }
              }
            />
          </React.Fragment>
        </div>
      </section>
    </div>
  </main>
</React.Fragment>
`;
